apps/web/src/views/Swap/components/AdvancedSwapDetails.tsx

```diff
@@ -178,6 +184,8 @@ export const TradeSummary = memo(function TradeSummary({
           <Text fontSize="14px">{`${formatAmount(realizedLPFee, 4)} ${inputAmount?.currency?.symbol}`}</Text>
         </RowBetween>
       )}
+
+      {isPaymasterAvailable && gasTokenSelector}
```

Comment:
Can we do this condition check on the upper level? The `AdvancedSwapDetails` component doesn't have to know the context. It'll show the gas token selector if the props is given